examples/ios/SampleClient/AppDelegate.m

```diff
@@ -33,29 +35,50 @@ @implementation AppDelegate
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
   // Override point for customization after application launch.
+  [Alohalytics setDebugMode:YES];
+  [Alohalytics setup:@"http://localhost:8080/" withLaunchOptions:launchOptions];
+
+  UIDevice * device = [UIDevice currentDevice];
+  [Alohalytics logEvent:@"deviceInfo" withKeyValueArray:@[@"deviceName", device.name, @"systemName", device.systemName, @"systemVersion", device.systemVersion]];
+
+  // Used for example purposes only to upload statistics in background.
+  [application setMinimumBackgroundFetchInterval:UIApplicationBackgroundFetchIntervalMinimum];
+
   return YES;
 }
 
 - (void)applicationWillResignActive:(UIApplication *)application {
   // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
   // Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
+  [Alohalytics logEvent:@"$willResignActive"];
 }
 
 - (void)applicationDidEnterBackground:(UIApplication *)application {
   // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
   // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+  [Alohalytics logEvent:@"$didEnterBackground"];
 }
 
 - (void)applicationWillEnterForeground:(UIApplication *)application {
   // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
+  [Alohalytics logEvent:@"$willEnterForeground"];
 }
 
 - (void)applicationDidBecomeActive:(UIApplication *)application {
   // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+  [Alohalytics logEvent:@"$didBecomeActive"];
 }
 
 - (void)applicationWillTerminate:(UIApplication *)application {
   // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+  [Alohalytics logEvent:@"$willTerminate"];
+}
+
+// You need to set "fetch" in UIBackgroundsModes in your plist.
+-(void)application:(UIApplication *)application performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler
+{
+  [Alohalytics forceUpload];
```

Comment:
Reminder: we spoke of `forceUpload` being the wrong call to make if you need the exponential delays to be respected.